test/ng/filter/filterSpec.js

```diff
@@ -425,6 +425,20 @@ describe('Filter: filter', function() {
       toThrowMinErr('filter', 'notarray', 'Expected array but received: {"toString":null,"valueOf":null}');
   });
 
+  it('should not throw an error if used with an array like object', function() {
```

Comment:
This test is obviously not sufficient. We need to test with "less array-like" objects.